compiler/modules/front/codeGen.bal

```diff
@@ -1229,6 +1245,108 @@ function codeGenExpr(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:Ex
     panic err:impossible("unrecognized expression type in code gen: " +  s:exprToString(expr));
 }
 
+function intOperand(CodeGenContext cx, bir:Operand operand, Range subExprRange) returns bir:IntOperand|CodeGenError {
+    if operand is bir:IntOperand {
+        return operand;
+    }
+    return cx.semanticErr("expected an int operand", subExprRange);
+}
+
+function exprRange(s:Expr expr) returns Range {
+    return { startPos: expr.startPos, endPos: expr.endPos };
+}
+
+function codeGenNilLiftResult(CodeGenContext cx, ExprEffect nonNilEffect, bir:BasicBlock? ifNilBlock, Position pos) returns ExprEffect {
+    if ifNilBlock == () {
+        return nonNilEffect;
+    }
+    else {
+        bir:BasicBlock block = cx.createBasicBlock();
+        bir:Register nonNilResult = <bir:Register> nonNilEffect.result;
```

Comment:
How do you know this cast cannot fail?

Reply:
`nonNilEffect` is the `ExprEffect` we are getting from the non lifted operation (such as `codeGenArithmeticBinaryExpr`) and they always create a temporary register to store their result (which we don't codeGen to be a constant value).